dragonfly/implementations/daq_run_interface.py

```diff
@@ -108,18 +125,34 @@ def start_run(self, run_name):
         self.run_name = run_name
         self._run_meta = {'DAQ': self.daq_name,
                          }
+        self._prerun_snapshot = {}
         self._do_prerun_gets()
         if not self._debug_without_meta_broadcast:
             self._send_metadata()
+        if not self._debug_without_snapshot_broadcast:
+            self._send_snapshot(snap_flag='pre')
         logger.debug('these meta will be {}'.format(self._run_meta))
         logger.info('start_run finished')
 
     def _do_prerun_gets(self):
-        logger.info('doing prerun meta-data gets')
-        result = self.provider.get(self._metadata_state_target, timeout=120)
-        these_metadata = result['value_raw']
+        logger.info('doing prerun meta-data and snapshot gets')
+        meta_result = self.provider.get(self._metadata_state_target, timeout=120)
+        these_metadata = meta_result['value_raw']
         self._run_meta.update(these_metadata)
-        self.determine_RF_ROI()
+        for target,item_list in self._snapshot_target_items.items():
+            snapshot_result = self.provider.cmd(target, 'get_latest', [self._start_time,item_list], timeout=120)
+            these_snaps = snapshot_result['value_raw']
+            self._prerun_snapshot.update(these_snaps)
+        if not self._debug_without_rf_roi:
```

Comment:
Right, the rf_roi info should only be relevant to include once per run and it's most appropriate to send it with the metadata file (that's created before the run in _do_prerun_gets) for the reasons stated by Ben. Line 155 is only populating the post-run snapshot of the slow control with the values in the endpoint_{numeric/string}_data tables of the db logged during the run.